### DISCUSSION TITLE
Ring styles don't render correctly on links due to Safari bug

### DISCUSSION THREAD

**Discussion by sentience** [tailwindlabs/tailwindcss#10238] [Help]
The box-shadow based [ring styles](https://tailwindcss.com/docs/ring-width) recently added to Tailwind do not seem to render properly on pseudo-class states (such as `hover:`) of inline elements (such as hyperlinks) in the latest version of Safari on macOS. Portions of the ring that extend beyond the bounds of the surrounding block element do not paint reliably when the box-shadow is added/removed.

In this screenshot, the top of the ring is missing:

<img width="660" alt="CleanShot 2023-01-04 at 13 59 11@2x" src="https://user-images.githubusercontent.com/89772/210476245-4d0e3f79-a70a-4292-9c86-a8e2cca1e01b.png">

(Note: this is not an `overflow: hidden` issue. Setting `overflow-visible` on the relevant elements does not help.)

An example of the Tailwind styles to reproduce this issue:

```html
<a href="focus:outline-none focus:ring focus:ring-red-600">Example</a>
```

I've been able to work around this issue by adding a large, transparent outline:

```html
<a href="focus:outline-none focus:outline-8 focus:ring focus:ring-red-600">Example</a>
```

As this appears to be a Safari rendering issue, I have reported this [here](https://bugs.webkit.org/show_bug.cgi?id=250059).

[Codepen demo of the Safari rendering issue](https://codepen.io/kevinyank/pen/rNrLEYa)

**Comment by vincentwinkel** [comment]
I got the same issue on Safari:


<img width="800" alt="Screenshot 2023-03-14 at 09 22 28" src="https://user-images.githubusercontent.com/1402505/224939527-7aaa7c87-ae5f-4ced-bbd9-b82a51d5d16c.png">
